test.battery_charge_ranges.py

```diff
@@ -0,0 +1,14 @@
+import unittest
+import battery_charge_ranges
+
+class test_battery_charge_ranges(unittest.TestCase):
+    def test_charge_ranges_no(self):
+        ranges = battery_charge_ranges.charge_ranges([1,2,3])

```

Comment:
This is a good strategy. You have written a simple single-range input and asserted one portion of the result (the length). And then you've written a test with 2 ranges and asserted more.
Suggestion: make the second test as the smallest possible delta on the first. Currently, the first test has single range and only length assert. Second test brings double range AND a deeper assert as well. To satisfy this you'll need to code the entire remaining functionality.
Why not make the second test still have the simple input but assert the content of the range? Then code for it and then move to multiple ranges.
Final advice: don't keep extending the same function. Whenever you need to satisfy a new test, make a new function by default. When that's not possible, extend the existing one.